jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -1802,34 +1874,32 @@ ecma_op_typedarray_define_own_property (ecma_object_t *obj_p, /**< TypedArray ob
 
       ecma_typedarray_info_t info = ecma_typedarray_get_info (obj_p);
 
-      if (array_index >= info.length)
+      if (!ecma_op_is_integer (num)
+          || num >= info.length
+          || num < 0
+          || (ecma_number_is_negative (num) && ecma_number_is_zero (num)))
       {
-        return ECMA_REJECT ((property_desc_p->flags & ECMA_PROP_SHOULD_THROW), "Invalid typed array index");
+        return ECMA_VALUE_FALSE;
       }
 
       if (property_desc_p->flags & ECMA_PROP_IS_VALUE_DEFINED)
       {
-        lit_utf8_byte_t *src_buffer = info.buffer_p + (array_index << info.shift);
-        ecma_value_t set_element = ecma_set_typedarray_element (src_buffer, property_desc_p->value, info.id);
+
+        lit_utf8_byte_t *src_buffer = info.buffer_p + ((uint32_t) num << info.shift);
+        ecma_value_t set_element = ecma_set_typedarray_element (src_buffer,
+                                                                info.array_buffer_p,
+                                                                property_desc_p->value,
+                                                                num, info.length, info.id);
```

Comment:
Each argument should go in different line. This is a lot of arguments though, could its number be reudced?

Reply:
I suggest to refactor set/get element for the following stucture:

get(ta_info *, index); set(ta_info*, index, value)